### ISSUE TITLE
`native-image-configure generate` emits `StringIndexOutOfBoundsException`

### ISSUE THREAD

**Issue by trustin** [oracle/graal#6980]
**Describe the issue**

I get multiple `Warning: Error processing trace entry ... StringIndexOutOfBoundsException` messages when I try to generate native image config from a trace file.

**Steps to reproduce the issue**

1. Download [core.json.gz](https://github.com/oracle/graal/files/12027684/core.json.gz) and [caller-filter.json.gz](https://github.com/oracle/graal/files/12027688/caller-filter.json.gz).
2. `gunzip core.json.gz caller-filter.json.gz`
3. `native-image-configure generate --trace-input=core.json --output-dir=native-image-config --caller-filter-file=caller-filter.json`

**Describe GraalVM and your environment:**

```
❯ ~/.asdf/installs/java/graalvm-22.3.2+java17/bin/java -version
openjdk version "17.0.7" 2023-04-18
OpenJDK Runtime Environment GraalVM CE 22.3.2 (build 17.0.7+7-jvmci-22.3-b18)
OpenJDK 64-Bit Server VM GraalVM CE 22.3.2 (build 17.0.7+7-jvmci-22.3-b18, mixed mode, sharing)

❯ uname -a               
Linux manjaro 5.15.85-1-MANJARO-ARM #1 SMP PREEMPT Wed Dec 21 18:18:45 UTC 2022 aarch64 GNU/Linux
```

**More details**

`generate` and `process-trace` produce the same error.

One of the error messages (they are more or less same):

```
Warning: Error processing trace entry {caller_class=net.bytebuddy.description.type.TypeDescription$ForLoadedType, result=false, args=[], function=getEnclosingMethod, tracer=reflect}: java.lang.StringIndexOutOfBoundsException: begin 0, end -1, length 5
	at java.base/java.lang.String.checkBoundsBeginEnd(String.java:4602)
	at java.base/java.lang.String.substring(String.java:2705)
	at org.graalvm.nativeimage.configure/com.oracle.svm.configure.trace.ReflectionProcessor.addFullyQualifiedDeclaredMethod(ReflectionProcessor.java:274)
	at org.graalvm.nativeimage.configure/com.oracle.svm.configure.trace.ReflectionProcessor.processEntry(ReflectionProcessor.java:228)
	at org.graalvm.nativeimage.configure/com.oracle.svm.configure.trace.TraceProcessor.processEntry(TraceProcessor.java:86)
	at org.graalvm.nativeimage.configure/com.oracle.svm.configure.trace.TraceProcessor.processTrace(TraceProcessor.java:62)
	at org.graalvm.nativeimage.configure/com.oracle.svm.configure.trace.TraceProcessor.process(TraceProcessor.java:57)
	at org.graalvm.nativeimage.configure/com.oracle.svm.configure.ConfigurationTool.generate(ConfigurationTool.java:308)
	at org.graalvm.nativeimage.configure/com.oracle.svm.configure.ConfigurationTool.main(ConfigurationTool.java:103)
```

**Comment by trustin** [comment]
I tried [GraalVM CE 23.1.0-dev-20230616_1155](https://github.com/graalvm/graalvm-ce-dev-builds/releases/tag/23.1.0-dev-20230616_1155) just in case it's fixed in dev, but I couldn't confirm anything because I get a seemingly unrelated error:

```
java.lang.IllegalAccessError: class com.oracle.svm.configure.trace.TraceProcessor (in module org.graalvm.nativeimage.configure) cannot access class com.oracle.svm.util.LogUtils (in module org.graalvm.nativeimage.base) because module org.graalvm.nativeimage.base does not export com.oracle.svm.util to module org.graalvm.nativeimage.configure
	at org.graalvm.nativeimage.configure/com.oracle.svm.configure.trace.TraceProcessor.processEntry(TraceProcessor.java:103)
	at org.graalvm.nativeimage.configure/com.oracle.svm.configure.trace.TraceProcessor.processTrace(TraceProcessor.java:64)
	at org.graalvm.nativeimage.configure/com.oracle.svm.configure.trace.TraceProcessor.process(TraceProcessor.java:59)
	at org.graalvm.nativeimage.configure/com.oracle.svm.configure.command.ConfigurationGenerateCommand.generate(ConfigurationGenerateCommand.java:294)
	at org.graalvm.nativeimage.configure/com.oracle.svm.configure.command.ConfigurationGenerateCommand.apply(ConfigurationGenerateCommand.java:61)
	at org.graalvm.nativeimage.configure/com.oracle.svm.configure.ConfigurationTool.main(ConfigurationTool.java:84)
```

**Comment by trustin** [comment]
Thoghts, anyone?

**Comment by oubidar-Abderrahim** [comment]
Thank you for reporting this issue, we'll take a look into it shortly.

Tracked internally on GR 47406

**Comment by peter-hofer** [comment]
Fixed in #7667. Thank you very much for reporting!
